Kernel/Net/Socket.h

```diff
@@ -95,7 +95,7 @@ class Socket : public File {
     timeval m_receive_deadline { 0, 0 };
     timeval m_send_deadline { 0, 0 };
 
-    Vector<RefPtr<Socket>> m_pending;
+    NonnullRefPtrVector<Socket> m_pending;
```

Comment:
The old vector was already keeping them alive, `RefPtr` and `NonnullRefPtr` both do the same job of that.
However I still think this change is good, since we're removing the ability to put null values in `m_pending` :)